compiler/natives/src/reflect/reflect.go

```diff
@@ -1325,6 +1325,26 @@ func getJsTag(tag string) string {
 	return ""
 }
 
+func (v Value) grow(n int) {
+	if n < 0 {
+		panic(`reflect.Value.Grow: negative len`)
+	}
+
+	s := v.object()
+	len := s.Get(`$length`).Int()
+	if len+n < 0 {
+		panic(`reflect.Value.Grow: slice overflow`)
+	}
+
+	cap := s.Get(`$capacity`).Int()
+	if len+n > cap {
```

Comment:
Do we need a `if cap+n < 0 {` overflow check, too?

Reply:
I based this update off of the original code...
```Go
func (v Value) grow(n int) {
	p := (*unsafeheader.Slice)(v.ptr)
	switch {
	case n < 0:
		panic("reflect.Value.Grow: negative len")
	case p.Len+n < 0:
		panic("reflect.Value.Grow: slice overflow")
	case p.Len+n > p.Cap:
		t := v.typ.Elem().(*rtype)
		*p = growslice(t, *p, n)
	}
}
```
...but it is always better to check our assumptions. So, here is my thought train (ramblings to myself):

We know:
- $len \le cap$, so we could say $len + x = cap$ where $x \ge 0$.
- $len + n \gt cap$ otherwise we won't grow, so the smallest $n$ that will still cause a grow is $cap - len + 1$.
- To have a negative value we must roll over past a $max$ value.

Okay, if we get a big $x$ by setting $len = 0$, we only need $n = cap + 1$ to grow.
If we set $cap = \frac{max}{2}$ (or any value smaller than $max$ but still with room to grow),
Then $len + n = 0 + (cap +1) = cap + 1 \lt max$
But $cap + n = cap + (cap + 1) = 2 cap + 1 = 2 \frac{max}{2} + 1 = max + 1 \gt max$.
Therefore, we can have a $len + n$ which isn't too big but still more than $cap$, and a $cap + n$ that is too big. Since we grow to $len + n$ we would be getting a false positive by checking $cap + n$.

I'm not sure if that stroll through my thoughts make sense (I can try to clean it up if you'd like), but it seems like we don't need to check `cap+n < 0` since `cap + n` may be much larger than `len + n` and we are only growing to `len + n`.